hkm/finna.py

```diff
@@ -22,7 +22,44 @@ def __init__(self, organisation='"0/HKM/"', material_type='"0/Image/"'):
         self.organisation = organisation
         self.material_type = material_type
 
-    def search(self, search_term, page=1, limit=20, language='fi', detailed=False):
+    def get_facets(self, search_term, language='fi', date_from=None, date_to=None):
+        url = FinnaClient.API_ENDPOINT + 'search'
+        payload = {
+            'lookfor': search_term,
+            'filter[]': ['format:' + self.material_type, 'building:' + self.organisation, 'online_boolean:"1"'],
+            'limit': 0,
+            'lng': language,
+            'facet[]': ['author_facet', 'collection', 'genre_facet', 'main_date_str', 'category_str_mv']
+        }
+        if date_from or date_to:
+            payload['filter[]'].append('search_daterange_mv:"[%s TO %s]"' % (date_from, date_to))
+            payload['search_daterange_mv_type'] = "within"
+
+        try:
+            r = requests.get(url, params=payload, timeout=self.timeout)
+        except requests.exceptions.RequestException:
+            LOG.error('Failed to communicate with Finna API', exc_info=True)
+            return None
+        else:
+            try:
+                r.raise_for_status()
+                # raise requests.exceptions.HTTPError()
+            except requests.exceptions.HTTPError:
+                LOG.error('Failed to communicate with Finna API', exc_info=True,
+                          extra={'data': {'status_code': r.status_code, 'response': repr(r.text)}})
+                return None
+
+        result_data = r.json()
+        if not 'status' in result_data or result_data['status'] != 'OK':
```

Comment:
Can the response status really be something else than 200, but the result still be NOT OK? Because otherwise errors are already raised and None is returned.

Reply:
I'm not really sure, this same logic is used in all the other queries to finna as well. I'll take a look if I can remove figure out the logic behind this.